src/Relation/Nullary/Finite/Setoid.agda

```diff
@@ -0,0 +1,148 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Notions of finiteness for setoids
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --cubical-compatible #-}
+
+module Relation.Nullary.Finite.Setoid where
+
+open import Data.Fin.Base using (Fin)
+open import Data.Nat.Base using (ℕ)
+open import Data.Product.Base using (_×_; _,_; swap; Σ-syntax; proj₁; ∃)
+open import Data.Unit.Base using (⊤)
+open import Function.Base using (id; _∘_)
+open import Function.Bundles using (Func; Injection; Surjection; Inverse)
+open import Level using (Level; _⊔_; 0ℓ) renaming (suc to lsuc)
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Bundles using (Setoid)
+open import Relation.Binary.Structures using (IsEquivalence)
+import Relation.Binary.Construct.On as On
+import Relation.Binary.PropositionalEquality.Properties as ≡
+
+private
+  variable
+    c ℓ c′ ℓ′ : Level
+
+Ω : ∀ p → Setoid (lsuc p) p
+Ω p = record
+  { Carrier = Set p
+  ; _≈_ = λ P Q → (P → Q) × (Q → P)
+  ; isEquivalence = record
+    { refl = id , id
+    ; sym = swap
+    ; trans = λ (f , g) (f′ , g′) → f′ ∘ f , g ∘ g′
+    }
+  }
+
+Subset : Setoid c ℓ → (p : Level) → Set (c ⊔ ℓ ⊔ lsuc p)
```

Comment:
Surely something like these two (`Subset` and `FullSubset`) already exist in stdlib elsewhere? Or should, independently.

Reply:
There is `Data.Fin.Subset` but the definition is very different.

Reply:
The names might be quite different.